numba/tests/test_np_functions.py

```diff
@@ -3621,6 +3625,49 @@ def test_cross2d_exceptions(self):
             str(raises.exception)
         )
 
+    def test_trim_zeros(self):
+
+        def arrays():
+            yield np.array([])
+            yield np.zeros(5)
+            yield np.zeros(1)
+            yield np.array([1, 2, 3])
+            yield np.array([0, 1, 2, 3])
+            yield np.array([0, 1, 2, 0, 0])
```

Comment:
Could we test arrays that have non-zero values that are not ints (e.g. `np.array([0., 1., 2., np.nan, 0.])`)?
We could also test other data types that work with the numpy version (e.g. `np.array(["Hello", "world"])`).